opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -49,160 +47,118 @@ class Meter:
-
-        Returns: A new `FloatCounter`
-        """
-
-    def create_int_counter(
-        self,
-        name: str,
-        description: str,
-        unit: str,
-        label_keys: LabelKeys,
-        span_context: SpanContext = None,
-    ) -> "IntCounter":
-        """Creates a counter type metric that contains int values.
-
-        Args:
-            name: The name of the counter.
-            description: Human readable description of the metric.
-            unit: Unit of the metric values.
-            label_keys: list of keys for the labels with dynamic values.
-                Order of the list is important as the same order MUST be used
-                on recording when suppling values for these labels.
-            span_context: The `SpanContext` that identifies the `Span`
-                that the metric is associated with.
-
-        Returns:
-            A new `IntCounter`
-        """
-
-    def create_float_gauge(
-        self,
-        name: str,
-        description: str,
-        unit: str,
-        label_keys: LabelKeys,
-        span_context: SpanContext = None,
-    ) -> "FloatGauge":
-        """Creates a gauge type metric that contains float values.
-
-        Args:
-            name: The name of the counter.
-            description: Human readable description of the metric.
-            unit: Unit of the metric values.
-            label_keys: list of keys for the labels with dynamic values.
-                Order of the list is important as the same order MUST be used
-                on recording when suppling values for these labels.
-            span_context: The `SpanContext` that identifies the `Span`
-                that the metric is associated with.
-
-        Returns:
-            A new `FloatGauge`
-        """
-
-    def create_int_gauge(
-        self,
-        name: str,
-        description: str,
-        unit: str,
-        label_keys: LabelKeys,
-        span_context: SpanContext = None,
-    ) -> "IntGauge":
-        """Creates a gauge type metric that contains int values.
-
-        Args:
-            name: The name of the counter.
-            description: Human readable description of the metric.
-            unit: Unit of the metric values.
-            label_keys: list of keys for the labels with dynamic values.
-                Order of the list is important as the same order MUST be used
-                on recording when suppling values for these labels.
-            span_context: The `SpanContext` that identifies the `Span`
-                that the metric is associated with.
-
-        Returns:
-            A new `IntGauge`
-        """
+    # TODO: RecordBatch
+
+
+class ValueType(Enum):
+    FLOAT = 0
+    INT = 1
+
+
+def create_counter(
+    name: str,
+    description: str,
+    unit: str,
+    value_type: "ValueType",
+    bidirectional: bool = False,
+    label_keys: List[str] = None,
+    span_context: SpanContext = None,
+) -> Union["FloatCounter", "IntCounter"]:
```

Comment:
It's a bit weird for the metric creation methods to return different types based on `value_type` instead of using a single generic type here, but neither solution is beautiful.

Reply:
You're right. It would actually be better to have one generic create method, but with the introduction of `bidirectional`, the metric types actually have slightly different behaviors.